wire-ios-data-model/Source/MLS/MLSService.swift

```diff
@@ -304,11 +308,34 @@ public final class MLSService: MLSServiceInterface {
         staleKeyMaterialDetector.keyingMaterialUpdated(for: groupID)
     }
 
+    public func createSelfGroup(for groupID: MLSGroupID) {
+        guard let context = context else {
+            return
+        }
+        do {
+            try createConversation(for: groupID)
+
+            let selfUser = ZMUser.selfUser(in: context)
+            let mlsSelfUser = MLSUser(from: selfUser)
+            Task {
+                do {
+                    try await addMembersToConversation(with: [mlsSelfUser], for: groupID)
+                } catch MLSAddMembersError.noInviteesToAdd {
+                    WireLogger.mls.debug("createConversation noInviteesToAdd, updateKeyMaterial")
+                    try await updateKeyMaterial(for: groupID)
+                }
+            }
+        } catch {
+            WireLogger.mls.error("create group for self conversation failed: \(error.localizedDescription)")
+        }
+    }
```

Comment:
This almost relies completely on other protocol requirements (the exception being `updateKeyMaterial`. If we made `updateKeyMaterial` part of the protocol then this method wouldn't need to be a requirement itself but could simply be an extension to the protocol.

Reply:
ok why not, but what's the benefit of it? Not have to implement it on each implementation

Reply:
so actually, not so good idea, since after I am not able to mock this method when using mockMLSService nor wait for it's completion.